src/activities/api/activities.js

```diff
@@ -37,6 +37,13 @@ export default {
     return this.list({ feedback_possible: true, group: groupId })
   },
 
+  icsUrl (filter) {
+    return axios.getUri({
+      params: filter,
+      url: window.location.origin + '/api/activities/ics/',
```

Comment:
oh, sorry. Then it should be like this?
```js
const backend = process.env.MODE === 'cordova' ? process.env.KARROT.BACKEND : ''

icsUrl (filter) {
  return axios.getUri({
    params: filter,
    url: backend + '/api/activities/ics/',
  })
},
```

Why don't other api calls require this too?

Reply:
They do :) see https://github.com/yunity/karrot-frontend/blob/c69420cde65416f7d5e70a0081e12289c12e4456/src/utils/cordova/setBaseURL.js#L15 (the is is loaded via the Cordova boot file)

... but now I noticed the call to `axios.getUri` which I missed before, so maybe given we set the baseURL for the axios instance it'll all be fine with just the path component here.

Reply:
Just tested dev.karrot.world via Saucelabs Android Emulator. Works for me. ICS file got downloaded:
<img width="402" alt="Bildschirmfoto 2021-07-07 um 16 30 41" src="https://user-images.githubusercontent.com/23213583/124777969-e0f53780-df40-11eb-9f3b-18c8dd537335.png">

I'm a bit confused right now. On dev it uses this on a single activity item: https://github.com/amengsk/karrot-frontend/blob/bf2430eeb6fad6bf36ee4372153029a98e0fff86/src/activities/components/ActivityItem.vue#L213
```js
`${window.location.origin}/api/activities/${this.activity.id}/ics/`\`
```

But with the interceptor, it should just work without the `window.location.origin`? Or isn't it intercepted because it's inside a computed property vs. api method like here?

And, it is failing on your phone? Maybe a `works on my machine` problem?  :)

Reply:
It's specific to the Cordova app version. The app includes all the asset files and the web view bit that loads the assets loads then via a file:// protocol, so without the full URL they are relative to where the assets got loaded from. Which is fine for the normal browser usage, but not for the app.

Stuff that goes via the general axios instance should be ok, as we set the baseURL there. But haven't ever used the getUri function myself so can't be sure there. 

If it's still not clear how it should be I can dig in a bit deeper later. Just on my phone right now. 

Thanks!

Reply:
Ah, I just see, there is no axios involved at the individual item🤦🏻‍♂️, just a simple `a href`. So this seems correct.
And for the part of this PR, it should be:
```js
icsUrl (filter) {
  return axios.getUri({
    params: filter,
    url: '/api/activities/ics/',
  })
},
```
So the interceptor should take care...

Would be great if you could have a look.
Is there a branch deployment for this PR?

Reply:
Sorry, my _I can dig in a bit deeper later_ got a bit more delayed than expected! Still high up my _karrot todo list_ though!

Reply:
I had a look, it currently looks like this in cordova:

![image](https://user-images.githubusercontent.com/4410802/125431360-29d9cddd-8f5a-442c-af11-73d009c28b4f.png)

Apparently, axios `getUri` doesn't use `baseURL` (yet): https://github.com/axios/axios/issues/2468

This might be a workaround for now:

```
icsUrl (filter) {
    const backend = process.env.MODE === 'cordova' ? process.env.KARROT.BACKEND : ''

    return axios.getUri({
      params: filter,
      url: backend + '/api/activities/ics/',
    })
  },
```

But UX isn't great, because those links get opening in a separate browser, where the user isn't logged in. So they get a prompt to authenticate themselves with http basic auth, which isn't that great...

<img src="https://user-images.githubusercontent.com/4410802/125436006-25710b58-bec4-4d0a-b0b2-02f125195441.png" height="400">

By the way, a similar problem affects the new invitation link and the ics link in ActivityItem, both in `master` or already in production, so it might be worth opening an issue about it:
https://github.com/yunity/karrot-frontend/blob/badd01286c495b87c28d18212f4b6e7906652a06/src/users/pages/Members.vue#L126-L128
https://github.com/yunity/karrot-frontend/blob/309a6f0407d6519b5683f5eca82a0a51028cf4f1/src/activities/components/ActivityItem.vue#L210-L213

Reply:
I think minimum bar for merging this is to fix the URLs so they at least resolve (could sneakily fix the invitation link one whilst at it...). Given it appears in a few places, maybe a utility function `absoluteUrl(path)`, or similar, could be handy.

And maybe we can deal with the issue about ICS download on cordova seperately (I think this is a bigger change, and it would be usable as it is).

Reply:
Are you still up for working on this @amengsk (or @larzon83)?

Reply:
So it should be `backend + ...` everywhere instead of `window.location.origin + ...`?

Reply:
I think it's not quite as simple, there are 2 kinds I think:
- **URLs to be clicked** - only need full URL in cordova (as clicking relative URL in browser automatically uses current origin)
- **URLs to be copied** - always need full URL (as you copy it outside the current context)

Reply:
Indeed, so something like this?

```js
const origin = process.env.MODE === 'cordova' ? process.env.KARROT.BACKEND : window.location.origin

    return axios.getUri({
      params: filter,
      url: origin + '/api/activities/ics/',
    })
```

Reminds me that I want to get rid of Cordova and switch to [TWA](https://developer.chrome.com/docs/android/trusted-web-activity/overview/).

Reply:
> URLs to be clicked - only need full URL in cordova

But it doesn't hurt? So we could use one utility function in all places:
```js
export const absoluteUrl = path => {
  const origin = process.env.MODE === 'cordova' ? process.env.KARROT.BACKEND : window.location.origin
  return `${origin}${path}`
}
```
Maybe also include a check, that the passed-in `path` starts with a `/`.

@tiltec Side note about TWAs: I think, one requirement is that Lighthouse performance score must be greater than 80. Currently, landingpage is about 50 and logged in about 30 (group wall) 😬